pyanaconda/modules/payloads/source/harddrive/initialization.py

```diff
@@ -0,0 +1,112 @@
+
+from pyanaconda.payload.utils import mount, unmount, PayloadSetupError
+from pyanaconda.payload.image import verify_valid_installtree, find_first_iso_image
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SetUpHardDriveSourceTask", "TearDownHardDriveSourceTask"]
+
+
+class TearDownHardDriveSourceTask(Task):
+    """Task to teardown installation source."""
+
+    def __init__(self, target_mount):
+        super().__init__()
+        self._device_mount = target_mount + "_device"
+        self._iso_mount = target_mount + "_iso"
+
+    @property
+    def name(self):
+        return "Tear down Hard drive installation source"
+
+    def run(self):
+        """Run live installation source un-setup."""
+        log.debug("Unmounting Hard drive installation source")
+        unmount(self._device_mount)
+        unmount(self._iso_mount)
+
+
+class SetUpHardDriveSourceTask(Task):
+    """Task to setup installation source."""
+
+    def __init__(self, target_mount, partition, directory):
+        # TODO need better name for target_mount
+        super().__init__()
+        self._device_mount = target_mount + "_device"
+        self._iso_mount = target_mount + "_iso"
+        self._partition = partition
+        self._directory = directory
+
+    @property
+    def name(self):
+        return "Set up Hard drive installation source"
+
+    def run(self):
+        """Run Hard drive installation source setup.
+
+        :return: Path to the install tree.
+        :rtype: str
+        """
+        log.debug("Setting up Hard drive source")
+
+        try:
+            matches = device_matches(self._partition)
+            if not matches:
+                raise SourceSetupError(
+                    "Device for Hard drive install {} does not exist".format(self._partition)
+                )
+
+            mount("/dev/" + matches[0], self._device_mount, "auto", "")
+
+        except (DeviceSetupError, MountFilesystemError, PayloadSetupError, SourceSetupError) as e:
+            log.error("Mount of partition device failed: %s", e)
+            raise SourceSetupError(str(e))
+
+        full_path_on_mounted_device = os.path.normpath(os.path.join(self._device_mount,
+                                                                    self._directory))
+        image_filename = find_first_iso_image(full_path_on_mounted_device)
+
+        if image_filename:
+            # there is a valid image
+            full_image_path = os.path.join(full_path_on_mounted_device, image_filename)
+            try:
+                mount(full_image_path, self._iso_mount, fstype='iso9660', options="ro")
+            except PayloadSetupError as e:
+                log.error("Mount of ISO file failed: %s", e)
+                unmount(self._device_mount)
+                raise SourceSetupError(str(e))
+
+            return self._iso_mount
```

Comment:
I would like to see more comments in this code or maybe the method could be split into several smaller ones. Somehow it is difficult to understand what the code does.